### PR TITLE
text replaced added

### PR DESCRIPTION
**Pull request by EugeneDar**

### CODE REVIEW

**Comment by EugeneDar** [comment]
@Graur please, have a look

File: src/main/eo/org/eolang/txt/text.eo [hunk 1]
```diff
@@ -69,3 +69,7 @@
 
   [i] > at
     ^.s.slice i 1 > @
+
+  # Returns string where all substrings
+  # substr1 changed to substr2
+  [substr1 substr2] > replaced /string
```
Comment (Graur) [inline src/main/eo/org/eolang/txt/text.eo:75]: @EugeneDar I think that the parameters of this object are incomprehensibly readable. We can call them `target` and `replacement`, or something similar. Got the idea?

File: src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public EOtext$EOreplaced(final Phi sigma) {
+        super(sigma);
+        this.add("substr1", new AtFree());
+        this.add("substr2", new AtFree());
+        this.add(
+            "φ",
+            new AtComposite(
+                this,
+                rho -> {
+                    final String oldsubstr = new Dataized(rho.attr("substr1").get())
```
Comment (Graur) [inline src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java:65]: @EugeneDar It is a compound name, can we make it shorter?

File: src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                        .take(String.class);
+                    final String newsubstr = new Dataized(rho.attr("substr2").get())
```
Comment (Graur) [inline src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java:67]: @EugeneDar Same here. Can we make `newsubstr` shorter?

File: src/test/eo/org/eolang/txt/text-tests.eo [hunk 4]
```diff
@@ -168,3 +168,30 @@
...
+    $.equal-to "abacaba"
+
+[] > text-replaced-3
+  assert-that > @
+    replaced.
+      text
+        "abacaba"
+      "abacaba"
+      "abacaba"
+    $.equal-to "abacaba"
```
Comment (Graur) [inline src/test/eo/org/eolang/txt/text-tests.eo:197]: @EugeneDar Let's use Russian or Unicode characters here. Or maybe add a new test

**Review by Graur — CHANGES_REQUESTED** [review]
@EugeneDar Why you implement it in Java instead of EO?

File: src/main/eo/org/eolang/txt/text.eo (hunk 1 above)
Comment (EugeneDar) [reply src/main/eo/org/eolang/txt/text.eo]: @Graur I agree, I'll fix it.

**Comment by EugeneDar** [comment]
> @EugeneDar Why you implement it in Java instead of EO?

@Graur we need implemented string methods soon, to use. For this reason, they will now be implemented in Java, but will of course be converted to EO in the future.

File: src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java (hunk 2 above)
Comment (EugeneDar) [reply src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java]: @Graur I'll fix it.

File: src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java (hunk 3 above)
Comment (EugeneDar) [reply src/main/java/EOorg/EOeolang/EOtxt/EOtext$EOreplaced.java]: @Graur I'll fix it.

File: src/test/eo/org/eolang/txt/text-tests.eo (hunk 4 above)
Comment (EugeneDar) [reply src/test/eo/org/eolang/txt/text-tests.eo]: @Graur I'll add it.

**Comment by Graur** [comment]
@EugeneDar then you can add `@todo`, and let [0pdd](https://www.yegor256.com/2017/04/05/pdd-in-action.html) bot to create such task for you. What do you think?

**Comment by EugeneDar** [comment]
@Graur I agree we should add `@todo` comment here. Can you please explane how to choose todo number?

**Comment by Graur** [comment]
@EugeneDar It is a number of the current task - 21

**Comment by EugeneDar** [comment]
Thanks

**Comment by Graur** [comment]
@EugeneDar your build failed

**Comment by EugeneDar** [comment]
@Graur my bad, please check again

File: src/main/eo/org/eolang/txt/text.eo [hunk 5]
```diff
@@ -69,3 +69,7 @@
 
   [i] > at
     ^.s.slice i 1 > @
+
+  # Returns string where all substrings
+  # target changed to replacement
+  [target substr2] > replaced /string
```
Comment (Graur) [inline src/main/eo/org/eolang/txt/text.eo:75]: @EugeneDar You forgot to replace `substr2` name. And I have no idea why all checks was passed successfully)

**Review by Graur — CHANGES_REQUESTED** [review]

File: src/main/eo/org/eolang/txt/text.eo (hunk 5 above)
Comment (EugeneDar) [reply src/main/eo/org/eolang/txt/text.eo:75]: @Graur perhaps we should notify yegor256 that this situation may arise?

Comment (EugeneDar) [reply src/main/eo/org/eolang/txt/text.eo:75]: @Graur it's very complex bug, can you please explain me how to explain this problem in issue?

Comment (Graur) [reply src/main/eo/org/eolang/txt/text.eo:75]: @EugeneDar Indeed, one of the parameters in `text.replaced` object have incorrect name, but all tests are passed successfully

Comment (Graur) [reply src/main/eo/org/eolang/txt/text.eo:75]: @EugeneDar I create a new issue https://github.com/objectionary/eo-strings/issues/29

Comment (Graur) [reply src/main/eo/org/eolang/txt/text.eo:75]: @yegor256 Сan you help us determine if this is a bug? We have different parameter's name in our `.eo` code and when we trying to dataize it in our `.java` code

Comment (Graur) [reply src/main/eo/org/eolang/txt/text.eo:75]: @EugeneDar let's fix this and merge your PR

**Review by Graur — CHANGES_REQUESTED** [review]

**Review by Graur — APPROVED** [review]
@yegor256 please, have a look

**Comment by EugeneDar** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@Graur check this one please

**Comment by Graur** [comment]
@yegor256 I have already approved these changes. Do you want me to merge them?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**